File: 面向对象程序设计（Java）/实验三 继承与接口/形状 ( Shape ).java

abstract class Shape {
    public abstract double getArea();

    public abstract double getVolume();
}

class Cylinder extends Shape {
    double r;
    double h;

    public Cylinder(double r, double h) {
        this.r = r;
        this.h = h;
    }

    @Override
    public double getArea() {
        return 2 * 3.14 * r * r + 2 * 3.14 * r * h;
    }

    @Override
    public double getVolume() {
        return 3.14 * r * r * h;
    }
}

class Ball extends Shape {
    double r;

    public Ball(double r) {
        this.r = r;
    }

    @Override
    public double getArea() {
        return 4.0 * 3.14 * r * r;
    }

    @Override
    public double getVolume() {
        return 4.0 / 3.0 * 3.14 * r * r * r;
    }
}

class Cuber extends Shape {
    double a;

    public Cuber(double a) {
        this.a = a;
    }

    @Override
    public double getArea() {
        return 6 * a * a;
    }

    @Override
    public double getVolume() {
        return a * a * a;
    }
}

class Cuboid extends Shape {
    double a, b, c;

    public Cuboid(double a, double b, double c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    @Override
    public double getArea() {
        return 2 * (a * b + a * c + b * c);
    }

    @Override
    public double getVolume() {
        return a * b * c;
    }
}

File: python应用开发/CSOJ/实验05-面向对象编程/形状 ( Shape ).py
class Shape:
    def getArea(self):
        pass

    def getVolume(self):
        pass


class Cylinder(Shape):
    def __init__(self, r, h):
        self.r = r
        self.h = h

    def getArea(self):
        return 2 * 3.14 * self.r * self.r + 2 * 3.14 * self.r * self.h

    def getVolume(self):
        return 3.14 * self.r * self.r * self.h


class Ball(Shape):
    def __init__(self, r):
        self.r = r

    def getArea(self):
        return 4.0 * 3.14 * self.r * self.r

    def getVolume(self):
        return 4.0 / 3.0 * 3.14 * self.r * self.r * self.r


class Cuber(Shape):
    def __init__(self, a):
        self.a = a

    def getArea(self):
        return 6 * self.a * self.a

    def getVolume(self):
        return self.a * self.a * self.a


class Cuboid(Shape):
    def __init__(self, a, b, c):
        self.a = a
        self.b = b
        self.c = c

    def getArea(self):
        return 2 * (self.a * self.b + self.a * self.c + self.b * self.c)

    def getVolume(self):
        return self.a * self.b * self.c


t = int(input())
for _ in range(t):
    r, h = map(float, input().split())
    cylinder = Cylinder(r, h)
    print(f"{cylinder.getArea():.2f} {cylinder.getVolume():.2f}")

    r = float(input())
    ball = Ball(r)
    print(f"{ball.getArea():.2f} {ball.getVolume():.2f}")

    a = float(input())
    cuber = Cuber(a)
    print(f"{cuber.getArea():.2f} {cuber.getVolume():.2f}")

    a, b, c = map(float, input().split())
    cuboid = Cuboid(a, b, c)
    print(f"{cuboid.getArea():.2f} {cuboid.getVolume():.2f}")